segmentstore/server/host/src/main/java/io/pravega/segmentstore/server/host/handler/PravegaRequestProcessor.java

```diff
@@ -594,4 +613,13 @@ private Long ignoreSegmentSealed(Throwable ex) {
 
         return null;
     }
+
+    /**
+     * Custom exception to indicate a {@link CancellationException} during a Read segment operation.
+     */
+    private class ReadCancellationException extends IllegalStateException {
```

Comment:
Why does this extend IllegalState? Reads being canceled is not an illegal state. This could extend CancellationException, or RuntimeException.

Reply:
`java.util.concurrent.CancellationException` extends IllegalStateException. Changed it to extend RuntimeException.